asset_classes/entity.py

```diff
@@ -0,0 +1,17 @@
+import math
+import os
+
+import pygame
+
+from utils.utils import load_png
+
+
+class Entity(pygame.sprite.Sprite):
+    def __init__(self, name, img, initial_width, initial_height, initial_angle):
+        pygame.sprite.Sprite.__init__(self)
+        self.name = name
+        self.img, self.rect = load_png(
+            img, initial_width, initial_height, initial_angle
+        )
+        self.screen = pygame.display.get_surface()
+        self.area = self.screen.get_rect()
```

Comment:
Next we implement the base class. We're abstracting away a lot more boilerplate here. Notice that it extends the Sprite class from pygame. So our Entity is a Sprite and has all the class methods and properties of a Sprite.

The tradeoff here is not all of our classes will be Entities because not all of them will be Sprites. This is okay, we might have a secondary base class. Or we can have one offs. 

Notice that we pass in an img property. It's actually just a filename. Then the load_png function from utils is going to set its width, height, and angle and return a Surface object.